bs/src/tablecloth.ml

```diff
@@ -462,13 +480,29 @@ module Tuple3 = struct
 
   let map_second = mapSecond
 
-  let mapThird ~(f : 'c -> 'z) ((a, b, c) : 'a * 'b * 'c) : 'a * 'b * 'z = (a, b, f c)
+  let mapThird = map
 
   let map_third = mapThird
 
   let mapEach ~(f : 'a -> 'x) ~(g : 'b -> 'y) ~(h : 'c -> 'z) ((a, b, c) : 'a * 'b * 'c) : 'x * 'y * 'z = (f a, g b, h c)
 
-  let map_each  = mapEach
+  let map_each = mapEach
+
+  let mapAll ~(f: 'a -> 'b) (a1, a2, a3) = (f a1, f a2, f a3)
+
+  let map_all = mapAll
+
+  let swirlLeft ((a, b, c) : 'a * 'b * 'c) : ('c * 'a * 'b) = (c, a, b)
```

Comment:
Good rename. I'm surprised at the signature. If it goes left, wouldn't we take the `first` element and move it to the back? `(a,b,c) -> (b,c,a)`? Is there a place that swirl comes from? If not maybe we should look at assembly and `rotate` which is a known thing at least.

Reply:
I thought the same thing! They seem backwards to me and I couldn't think of a way for it to make sense.

I will have a look.

Reply:
That's not an official or community package. Given that it hasn't had as much community review or design input as the other extra libs, no need to stick to closely to it.